modern/src/map/Map.js

```diff
@@ -35,6 +35,13 @@ const updateReadyValue = value => {
   readyListeners.forEach(listener => listener(value));
 };
 
+const onIdle = function () {
+  const zoom = Math.round(map.getZoom())
+  if (zoom !== map.getZoom()) {
+    map.zoomTo(zoom)
+  }
+};
+
 const initMap = async () => {
```

Comment:
Also, what happens when we change map layer?

Reply:
map layer changes -> mapReady becomes false -> clean up function triggers and event listener gets removed -> other map layer loads -> mapReady becomes true -> if it's a raster map event listener gets added again otherwise no action